WeakAurasOptions/AuthorOptions.lua

```diff
@@ -402,135 +477,207 @@ local function getArrayStr(option, array, index)
+        return
+      end
       for _, optionData in pairs(option.references) do
         local childData = optionData.data
         local childConfig = optionData.config
-        childConfig[option.key] = num
-        WeakAuras.Add(childData)
-      end
+        OptionsPrivate.Private.TimeMachine:Append({
+          uid = childData.uid,
+          actionType = "set",
+          path = expandUserPath(data, optionData.path),
+          payload = num
+        })
+      end
+      OptionsPrivate.Private.TimeMachine:Commit()
       WeakAuras.ClearAndUpdateOptions(data.id, true)
     end
   end
 end
 
+---@type fun(data: auraData, option: Option, key: string): fun(_, value: number)
 local function setColor(data, option, key)
   return function(_, r, g, b, a)
     local color = {r, g, b, a}
+    OptionsPrivate.Private.TimeMachine:StartTransaction()
     for id, optionData in pairs(option.references) do
       local childOption = optionData.options[optionData.index]
       local childData = optionData.data
-      childOption[key] = color
-      WeakAuras.Add(childData)
-    end
+      OptionsPrivate.Private.TimeMachine:Append({
+        uid = childData.uid,
+        actionType = "set",
+        path = expandPath(optionData.path, key),
+        payload = color
+      })
+    end
+    OptionsPrivate.Private.TimeMachine:Commit()
     WeakAuras.ClearAndUpdateOptions(data.id, true)
   end
 end
 
+---@type fun(data: auraData, option: Option): fun(_, r: number, g: number, b: number, a: number)
 local function setUserColor(data, option)
   return function(_, r, g, b, a)
     local color = {r, g, b, a}
+    OptionsPrivate.Private.TimeMachine:StartTransaction()
     for id, optionData in pairs(option.references) do
       local childData = optionData.data
       local childConfig = optionData.config
-      childConfig[option.key] = color
-      WeakAuras.Add(childData)
-    end
+      OptionsPrivate.Private.TimeMachine:Append({
+        uid = childData.uid,
+        actionType = "set",
+        path = expandUserPath(data, optionData.path),
+        payload = color
+      })
+    end
+    OptionsPrivate.Private.TimeMachine:Commit()
     WeakAuras.ClearAndUpdateOptions(data.id, true)
   end
 end
 
-local function setSelectDefault(data, option, key)
+---@type fun(data: auraData, option: Option): fun(_, value: number)
+local function setSelectDefault(data, option)
   return function(_, value)
     for id, optionData in pairs(option.references) do
       local childOption = optionData.options[optionData.index]
       local childData = optionData.data
-      childOption.default = min(value, #childOption.values)
-      WeakAuras.Add(childData)
-    end
+      OptionsPrivate.Private.TimeMachine:Append({
+        uid = childData.uid,
+        actionType = "set",
+        path = expandPath(optionData.path, "default"),
+        payload = min(value, #childOption.values)
+      })
+    end
+    OptionsPrivate.Private.TimeMachine:Commit()
```

Comment:
You removed the StartTransaction in 2d7a4cfe07401399295e8617cc7894506dde96de. Was that intentional?